src/TriangleNodesProvisioner.cpp

```diff
@@ -441,8 +444,116 @@ namespace blitzdg {
     }
     
     void TriangleNodesProvisioner::buildMaps() {
-        //const MeshManager& meshMgr = *Mesh2D;
-        // WIP here...
+        firstIndex ii;
+        secondIndex jj;
+        thirdIndex kk;
+
+        index_matrix_type nodeIds(NumLocalPoints, NumElements);
+        
+        const index_matrix_type& Fmsk = *Fmask;
+        index_vector_type& vmM = *vmapM;
+        index_vector_type& vmP = *vmapP;
+        index_vector_type& mP = *mapP;
+
+        const index_vector_type& E2E = Mesh2D.get_EToE();
+        const index_vector_type& E2F = Mesh2D.get_EToF();
+        const index_vector_type& E2V = Mesh2D.get_Elements(); 
+
+        real_matrix_type xmat(NumLocalPoints, NumElements, ColumnMajorOrder());
+        real_matrix_type ymat(NumLocalPoints, NumElements, ColumnMajorOrder());
+
+        xmat = *xGrid; ymat = *yGrid;
+
+        real_vector_type x(NumElements*NumLocalPoints);
+        real_vector_type y(NumElements*NumLocalPoints);
+        reshapeMatTo1D(xmat, x.data(), false);
+        reshapeMatTo1D(ymat, y.data(), false);
+
+        // Assemble global volume node numbering.
+        nodeIds = ii + NumLocalPoints*jj;
+
+        Array<index_type, 3> vmapM3(NumFacePoints, NumFaces, NumElements);
+        Array<index_type, 3> vmapP3(NumFacePoints, NumFaces, NumElements);
+        Array<index_type, 3> mapP3(NumFacePoints, NumFaces, NumElements);
```

Comment:
Would it be worthwhile to make a new 3D tensor type and add it to `Types.h`? Something like:
```
using index_tensor3_type = Array<index_type, 3>;
```